### PR TITLE
Update implementation to change deliveryId to session-scoped

### LINKED ISSUE #22: Include delivery id sequence number per message
Receiver expects each message to have an increasing `delivery-id` sequence number.

### PR DESCRIPTION
**Pull request by izzymoriguchi**
Fix #22

### CODE REVIEW

File: src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java [hunk 1]
```diff
@@ -795,17 +794,6 @@ private int decodeTransfer(
...
-            {
-                assert deliveryId != NO_DELIVERY_ID; // TODO: error
-                assert deliveryId == sender.deliveryId + 1; // TODO: error
-                sender.deliveryId = deliveryId;
-            }
-
-            if (deliveryId != NO_DELIVERY_ID && deliveryId != sender.deliveryId) // TODO: error
-            {
-                break decode;
-            }
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java:808]: Looks like we lost the check for invalid inbound `deliveryId`, this needs to be restored, right?
Please make sure to separate `remoteDeliveryId` coming from the client towards us from `deliveryId` generated by us, heading towards the client.

File: src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java [hunk 2]
```diff
@@ -2192,19 +2180,8 @@ private void onDecodeTransfer(
...
-                    {
-                        link.onDecodeError(traceId, authorization, LINK_TRANSFER_LIMIT_EXCEEDED);
-                    }
-                    else
-                    {
-                        link.onDecodeTransfer(traceId, authorization, reserved, deliveryId, deliveryTag, messageFormat, settled,
-                            resume, aborted, batchable, more, buffer, offset, limit);
-                    }
+                    link.onDecodeTransfer(traceId, authorization, reserved, deliveryTag, messageFormat, settled,
+                        resume, aborted, batchable, more, buffer, offset, limit);
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java:2195]: Do we need to keep these checks to validate inbound `deliveryId` or is it fully handled inside the link instead now?

File: src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java [hunk 3]
```diff
@@ -2674,11 +2656,11 @@ private void onApplicationDataInit(
                     final AmqpDataExFW dataEx = extension.get(amqpDataExRO::tryWrap);
                     assert dataEx != null;
 
+                    deliveryId++;
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java]: Suggest moving this just above the call to `onApplicationDataInit` in `onApplicationData`.

**Review by jfallows — CHANGES_REQUESTED** [review]

File: src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java (hunk 2 above)
Comment (izzymoriguchi) [reply src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java:2195]: I believe I handled it inside the link's `onDecodeTransfer`

**Review by jfallows — DISMISSED** [review]

**Review by jfallows — APPROVED** [review]

**Pull request merged**